slice/Ice/LocatorRegistry.ice

```diff
@@ -45,8 +45,8 @@ module Ice
         /// @throws AdapterNotFoundException Thrown when the locator only allows registered object adapters to register
         /// their endpoints and no object adapter with this adapter ID was registered with the locator.
         /// @throws AdapterAlreadyActiveException Thrown when an object adapter with the same adapter ID has already
-        /// registered its endpoints.
-        // Note: idempotent is not correct but kept for backwards compatibility with old implementations.
+        /// registered its endpoints. Since this operation is marked idempotent, this exception may be thrown when the
```

Comment:
From the Ice point of view, idempotent just means the Ice client runtime can or does retry when the connection drops while waiting for response. 

For this particular operation, it's arguably correct as far as the LocatorRegistry impl is concerned (e.g. IceGrid registry) but results in a very confusing behavior for the caller. The Ice runtime catches this exception and throws instead an ObjectAdapterIdInUse exception ... and the user scratches his head trying to figure out what happened.